docs/RFCS/replicate_queue_remove_replicas.md

```diff
@@ -0,0 +1,139 @@
+- Feature Name: Replicate Queue: Remove Replicas Enhancement
+- Status: draft
+- Start Date: 2015-08-18
+- RFC PR: #2153
+- Cockroach Issue: #620
+
+# Summary
+The Replicate Queue is currently used to add replicas to a range that is
+under-replicated. It can only add replicas, and it can only handle ranges with
+homogenous ReplicaAttrs (i.e. every replica requires the same attributes). This
+change will enhance the replicate queue to allow for down-replication: the
+removal of replicas from a range which has too many replicas. 
+
+With this addition, the replicate queue will be able to correct the replication
+of any range to its optimal replica count, regardless of how the range got into
+a sub-optimal state. Most immediately, this will provide crucial support for the
+highly desired "Rebalance" and "Repair" operations.
+
+This RFC will not address the issue of non-homogenous ReplicaAttrs. While not
+exceptionally difficult, we are simply not yet deploying with non-homogenous
+replicas and the work can be postponed.
+
+# Motivation
+Rebalancing is the major missing component from Cockroach at this point; without
+it, Cockroach is constrained to running on three stores. Rebalancing will
+opportunistically relocate replicas from over-utilized stores to under-utilized
+stores.
+
+Another important missing feature is Repair; detecting when a store is down, and
```

Comment:
it doesn't really, though. After reading the RFC I'm mostly wondering how the replicate queue will be used by repair and rebalance (after all, they do mostly the same thing, so they'll avoid duplicating all the logic), which are also queues. If you fleshed that out, it would make more sense.